src/api/request-hooks/request-logger.js

```diff
@@ -63,7 +63,7 @@ class RequestLoggerImplementation extends RequestHook {
         const loggerReq = this._internalRequests[event.requestId];
 
         if (!loggerReq)
-            throw new TypeError(`Cannot find a recorded request with id=${event.id}. This is an internal TestCafe problem. Please contact the TestCafe team and provide an example to reproduce the problem.`);
+            return;
```

Comment:
Could you please add a note here describing the situation?

Reply:
Also IMHO  it would be ideal to log when we don't have a request for the response to the debug log using the `debug` module.